internal/domain/entityintegrationmapping/model.go

```diff
@@ -0,0 +1,155 @@
+package entityintegrationmapping
+
+import (
+	"github.com/flexprice/flexprice/ent"
+	ierr "github.com/flexprice/flexprice/internal/errors"
+	"github.com/flexprice/flexprice/internal/types"
+)
+
+// EntityIntegrationMapping represents an entity integration mapping in the system
+type EntityIntegrationMapping struct {
+	// ID is the unique identifier for the mapping
+	ID string `db:"id" json:"id"`
+
+	// EntityID is the FlexPrice entity ID (e.g., customer_id, plan_id, etc.)
+	EntityID string `db:"entity_id" json:"entity_id"`
+
+	// EntityType is the type of entity (e.g., customer, plan, invoice, subscription, etc.)
+	EntityType string `db:"entity_type" json:"entity_type"`
+
+	// ProviderType is the payment provider type (e.g., stripe, razorpay, etc.)
+	ProviderType string `db:"provider_type" json:"provider_type"`
+
+	// ProviderEntityID is the provider's entity ID (e.g., stripe_customer_id, etc.)
+	ProviderEntityID string `db:"provider_entity_id" json:"provider_entity_id"`
+
+	// Metadata contains provider-specific data
+	Metadata map[string]interface{} `db:"metadata" json:"metadata"`
+
+	// EnvironmentID is the environment identifier
+	EnvironmentID string `db:"environment_id" json:"environment_id"`
+
+	types.BaseModel
+}
+
+// FromEnt converts an ent EntityIntegrationMapping to a domain EntityIntegrationMapping
+func FromEnt(e *ent.EntityIntegrationMapping) *EntityIntegrationMapping {
+	if e == nil {
+		return nil
+	}
+	return &EntityIntegrationMapping{
+		ID:               e.ID,
+		EntityID:         e.EntityID,
+		EntityType:       e.EntityType,
+		ProviderType:     e.ProviderType,
+		ProviderEntityID: e.ProviderEntityID,
+		Metadata:         e.Metadata,
+		EnvironmentID:    e.EnvironmentID,
+		BaseModel: types.BaseModel{
+			TenantID:  e.TenantID,
+			Status:    types.Status(e.Status),
+			CreatedAt: e.CreatedAt,
+			UpdatedAt: e.UpdatedAt,
+			CreatedBy: e.CreatedBy,
+			UpdatedBy: e.UpdatedBy,
+		},
+	}
+}
+
+// FromEntList converts a list of ent EntityIntegrationMapping to domain EntityIntegrationMapping
+func FromEntList(mappings []*ent.EntityIntegrationMapping) []*EntityIntegrationMapping {
+	result := make([]*EntityIntegrationMapping, len(mappings))
+	for i, e := range mappings {
+		result[i] = FromEnt(e)
+	}
+	return result
+}
+
+// ValidateEntityType validates the entity type
+func ValidateEntityType(entityType string) bool {
+	validTypes := map[string]bool{
+		"customer":     true,
+		"plan":         true,
+		"invoice":      true,
+		"subscription": true,
+		"payment":      true,
+		"credit_note":  true,
+	}
+	return validTypes[entityType]
+}
+
+// ValidateProviderType validates the provider type
+func ValidateProviderType(providerType string) bool {
+	validProviders := map[string]bool{
+		"stripe":   true,
+		"razorpay": true,
+		"paypal":   true,
+	}
+	return validProviders[providerType]
```

Comment:
create enums and move to types/ folder. Refer `type SubscriptionStatus string` as example